spec/controllers/admin_controller_spec.rb

```diff
@@ -0,0 +1,30 @@
+require 'rails_helper'
+
+RSpec.describe AdminController, type: :controller do
+  describe '#authenticate_admin' do
```

Comment:
After talking to Brendan about this we'd like to make use of RSpec's shared examples. What we really care about here is that the controllers that inherit from `AdminController` contain that functionality.

We don't want to have to write the same specs out every time for each controller that inherits from `AdminController`, though. Fortunately shared examples let's us define a series of assertions for the admin controller and simply include them in each controller that inherits from it.

```ruby
RSpec.describe SomeController do
  it_behaves_like "an admin controller"
end

RSpec.describe SomeOtherController do
  it_behaves_like "an admin controller"
end
```